torch/_inductor/codegen/triton.py

```diff
@@ -871,9 +871,30 @@ def index_expr(cls, expr, dtype):
 
     @staticmethod
     def masked(mask, body, other):
-        with V.kernel.mask_loads(mask) as new_mask:
+        handler = V.get_ops_handler()
+        found_non_load = False
+        load_count = 0
+
+        class FindLoad:
+            def __getattr__(self, name: str) -> Callable[..., CSEVariable]:
+                def inner(*args, **kwargs):
+                    nonlocal found_non_load, load_count
+                    if name != "load":
+                        found_non_load = True
+                    else:
+                        load_count += 1
+                    return getattr(handler, name)(*args, **kwargs)
+
+                return inner
```

Comment:
This is too complex. Just look at the FXgraph within `body` and check that the returned value is a load or a tuple of loads (perhaps with some extra logic to handle potential edge cases, like "no more loads than those fed to the output", no stores, etc).